### PR TITLE
Fix/null deletion edge case

### PR DESCRIPTION
**Pull request by JeremyWesthead**
Give `None` for gene position in cases where the start of a deletion in a revcomp gene goes past the start of the gene's promoter

### CODE REVIEW

**Comment by mcolpus** [comment]
What would be the impact of this for the katG deletion?

File: gumpy/difference.py [hunk 1]
```diff
@@ -239,6 +240,16 @@ def get_gene_pos(
...
 
+                # Edge case of deletion starting in revcomp gene and extending
+                # past gene start, so return None
+                if (
+                    self.genome2.stacked_nucleotide_number[
+                        self.genome2.stacked_gene_name == gene
+                    ][-1]
+                    > nc_num
+                ):
+                    return None
```
Comment (mcolpus) [inline gumpy/difference.py:251]: Is it easy to add a test case for this just to make sure there aren't any off by one errors, or something like that?

**Comment by JeremyWesthead** [comment]
> What would be the impact of this for the katG deletion?

This should result in 0 changes to the prediction, but should change the `variants` block
```
        "nucleotide_index": 2156034,
        "gene_name": "katG",
        "gene_position": null,
        "codon_idx": null,
```
This gives a null value for `gene_position` rather than a misleading position which doesn't exist

File: gumpy/difference.py (hunk 1 above)
Comment (JeremyWesthead) [reply gumpy/difference.py:251]: It's not particularly easy to add a test case for this - it'd either have to be a new dummy genome or a contrived TB test case (slow). However, it shouldn't be off by one as `>` should be checking that the calculated `nc_num` is less than the minimum gene position

**Review by mcolpus — APPROVED** [review]
